README.md

```diff
@@ -195,11 +195,114 @@ Subscribe to the topic published by the AWS IoT thing in the client, to see mock
 ![mqtt subscription](images/mqtt_subscription.png)
 You can now convert any ROS2 topic data and send it over as an MQTT topic over AWS IoT Core, with a thing transformation layer from ROS2 topic to JSON-formatted messages sent over an MQTT topic.
 
+## IoT Shadow Node
+
+This sample application requires the Telemetry MQTT sample application setup steps, as it reuses that IoT Thing and certificate. It shows how to create and interact with an IoT Core Named Shadow. The example node within subscribes to shadow changes and publishes them on a ROS topic, as well as exposing service calls for other ROS nodes to update the shadow.
+
+![Shadow Node Block Diagram](images/shadow_node.png)
+
+This node is demonstrated using a safe cracker robot, as shown above. The robot is turning a dial on a safe, swapping between clockwise and counter-clockwise. A random number generator updates the shadow with the digit to turn the dial to, and the safe cracker robot listens for shadow updates and turns the dial to that digit.
+
+The digit generator uses a service call to set the `desired` state. The shadow node listens to any shadow updates and publishes them on a ROS2 topic. The safe cracker robot subscribes to these updates, and uses a service call to set the `reported` state. More information on shadow use is available later in this file.
+
+To set up this application, you first need to create a named shadow for the Thing. This can be done as follows:
+
+```bash
+export SHADOW_NAME=my_ros2_shadow
+aws iot-data update-thing-shadow --thing-name $THING_NAME --shadow-name $SHADOW_NAME --payload "{\"state\":{\"reported\":{}}}" --cli-binary-format raw-in-base64-out /dev/null
```

Comment:
This assumes that the previous env vars are still present correct? Should we add a comment to say "without closing the terminal" or something that preserves the previous vars? This is a nice to have so feel free to make a judgement call.